src/front/wgsl/lexer.rs

```diff
@@ -197,16 +199,16 @@ impl<'a> Lexer<'a> {
                 Token::Arrow => "->",
                 Token::Unknown(_) => "unknown",
                 Token::UnterminatedString => "string",
+                Token::Trivia => "trivia",
                 Token::End => "",
             };
             Err(Error::Unexpected(token, description))
         }
     }
 
     pub(super) fn skip(&mut self, what: Token<'_>) -> bool {
-        let (token, rest) = consume_token(self.input);
-        if token == what {
-            self.input = rest;
+        if self.peek() == what {
+            let _ = self.next();
```

Comment:
This currently consumes the token twice but we could fix this before merging

Reply:
sounds good, let's try to fix this!

Reply:
I wasn't sure what to call it but we have `peek_token_and_rest` now – let me know if this is clear enough